runtime/caml/platform.h

```diff
@@ -42,6 +42,10 @@ Caml_inline void cpu_relax(void) {
 #elif defined(__riscv)
   /* Encoding of the pause instruction */
   __asm__ volatile (".4byte 0x100000F");
+#elif defined(__ppc64__)
+  __asm__ volatile ("or 1, 1, 1 # low priority");
+  __asm__ volatile ("or 2, 2, 2 # medium priority");
+  __asm__ volatile ("" ::: "memory");
```

Comment:
This looks right for POWER9, but I notice that POWER10 has a more conventional wait instruction ([see Linux patch](https://patchwork.ozlabs.org/project/linuxppc-dev/patch/20220720132132.903462-2-npiggin@gmail.com/#2939372))).  I don't have any POWER10 hardware to test this though, and it's not a blocker for this PR.

Reply:
Good find, I suppose we can do processor detection at configure time to emit a POWER10 specific instruction? Note this will support both Power 8, 9 and 10, which should cover the available supported machines.
Happy to test on POWER10 if someone has a loan machine :-)

Reply:
I've pinged our IBM contacts for POWER10 hardware, but think this should be addressed separately from this PR. The qemu power10 mode exists but I'm not a fan of using it as you can't really assess any realistic performance impact.